src/Common/ZooKeeper/ZooKeeper.cpp

```diff
@@ -271,6 +280,22 @@ Coordination::Error ZooKeeper::tryGetChildrenWatch(
     Coordination::Stat * stat,
     Coordination::WatchCallback watch_callback,
     Coordination::ListRequestType list_request_type)
+{
+    Coordination::Error code = getChildrenImpl(path, res, stat,
+        watch_callback ? std::make_shared<Coordination::WatchCallback>(watch_callback) : Coordination::WatchCallbackPtr{},
+        list_request_type);
+
+    if (!(code == Coordination::Error::ZOK || code == Coordination::Error::ZNONODE))
+        throw KeeperException(code, path);
+
+    return code;
+}
```

Comment:
Have you thought about just calling the other version with `WatchCallbackPtr` instead of duplicating the (honestly very simple) error handling code? Usually I think this kind of duplication is kind of okay, but as we can eliminate it without sacrificing anything in my opinion, I think it make sense to eliminate.

P.S.: I see that tha function above is also doing the same, so maybe this is okay, if this is fine, then feel free to leave it as is.